### PR TITLE
chore: update gradle, lib versions

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
Update gradle, bring in new build features, update libs.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-all.zip
```
Comment (tim-mwangi) [inline gradle/wrapper/gradle-wrapper.properties:3]: interesting! glad you did not run into issues!

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[versions]
```
Comment (tim-mwangi) [inline gradle/libs.versions.toml:1]: Yet another file format.

File: config-service/build.gradle.kts [hunk 3]
```diff
@@ -55,33 +55,25 @@ tasks.integrationTest {
...
 
   // Integration test dependencies
-  integrationTestImplementation("org.junit.jupiter:junit-jupiter:5.7.1")
-  integrationTestImplementation("com.google.guava:guava:30.1.1-jre")
-  integrationTestImplementation("org.yaml:snakeyaml:1.28")
-  integrationTestImplementation(project(":config-service-impl"))
-  integrationTestImplementation("org.hypertrace.core.serviceframework:integrationtest-service-framework:0.1.23")
-  integrationTestImplementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.4.0")
-  integrationTestImplementation("org.hypertrace.core.documentstore:document-store:0.5.7")
-  constraints {
```
Comment (tim-mwangi) [inline config-service/build.gradle.kts:77]: no more constraints?

**Review by tim-mwangi — APPROVED** [review]

File: config-service/build.gradle.kts (hunk 3 above)
Comment (aaron-steinfeld) [reply config-service/build.gradle.kts:77]: thought we had put those constraints into the framework, but it appears not - fixing.

**Review by tim-mwangi — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 2 above)
Comment (aaron-steinfeld) [reply gradle/libs.versions.toml:1]: tom's obvious minimal language

Comment (aaron-steinfeld) [reply gradle/libs.versions.toml:1]: (we've already checked these in elsewhere and there's nothing to really learn here but I was curious too - their arguments are at the end of this design doc https://github.com/gradle/gradle/files/5646826/2020-12-05-Central.declaration.of.dependencies.Shared.externally.pdf )

**Pull request merged**